### PR TITLE
Added register method realization and covered by JUnit tests

### PR DESCRIPTION
**Pull request by pkobevko**

### CODE REVIEW

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -2,13 +2,52 @@
...
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validate(user);
```
Comment (nechupara) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:16]: split this method into separate methods like validateLogin(user), validateAge(user) and so on.

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
```
Comment (nechupara) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:12]: I suggest you to initialize registrationService variable in @BeforeAll block

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setAge(18);
```
Comment (nechupara) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:18]: I suggest you to create MIN_AGE variable in RegistrationServiceImplTest class and use it throughout this class

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    @Test
+    void register_FiveCharactersLogin_notOk() {
+        user.setLogin("abcde");
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_SixCharactersLogin_Ok() {
```
Comment (nechupara) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:84]: maybe you should replace word "six" into "minLength" or something like this. But it's up to you

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    @Test
+    void register_FiveCharactersPassword_notOk() {
+        user.setPassword("abcde");
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_SixCharactersPassword_Ok() {
```
Comment (nechupara) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:122]: the same: maybe you should replace word "six" into "minLength" or something like this. But it's up to you

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    @Test
+    void register_under18YearsOldAge_notOk() {
+        user.setAge(16);
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_18YearsOldAge_Ok() {
```
Comment (nechupara) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:152]: and here I would suggest to replace number 18 to "minAge"

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+        user.setAge(18);
```
Comment (nechupara) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:153]: and use MIN_AGE variable instead of magic number.

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    void register_NegativeAge_notOk() {
+        user.setAge(-1);
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_under18YearsOldAge_notOk() {
+        user.setAge(16);
```
Comment (nechupara) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:145]: I would suggest to use edge case MIN_AGE - 1 (17) instead

**Review by nechupara — CHANGES_REQUESTED** [review]
Good job! But consider my suggestions )

**Review by nechupara — APPROVED** [review]
nice )

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+    @Test
+    void register_threeCharactersLogin_notOk() {
+        user.setLogin("abc");
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_fiveCharactersLogin_notOk() {
```
Comment (maxline) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:83]: I'd suggest not put a harcoded values in test methods names, if in in future the min length value be changed you will need rename the lot's of test methods names. 
```suggestion
    void register_loginIsOneUnderMinLength_notOk() {
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+    @Test
+    void register_minLengthLogin_Ok() {
+        user.setLogin("abcdef");
+        registrationService.register(user);
+        Assertions.assertTrue(Storage.people.contains(user));
+    }
+
+    @Test
+    void register_eightCharactersLogin_Ok() {
```
Comment (maxline) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:98]: the same reason as in previous comment:
```suggestion
    void register_loginIsOverMinLength_ok() {
```
Please fix all similar cases.
Also please be attentive to the small and capital letter ("ok"), template was described in common mistakes. Fix in all test names please.

**Review by maxline — CHANGES_REQUESTED** [review]
Well done! I left just a couple minor suggestions.

**Review by sarakhmen — APPROVED** [review]
Good job ;)
